src/modules/rover_ackermann/RoverAckermannControl/RoverAckermannControl.cpp

```diff
@@ -41,27 +41,19 @@ RoverAckermannControl::RoverAckermannControl(ModuleParams *parent) : ModuleParam
 {
 	updateParams();
 	_rover_ackermann_status_pub.advertise();
-	pid_init(&_pid_throttle, PID_MODE_DERIVATIV_NONE, 0.001f);
-
 }
 
 void RoverAckermannControl::updateParams()
 {
 	ModuleParams::updateParams();
 
-	pid_set_parameters(&_pid_throttle,
-			   _param_ra_speed_p.get(), // Proportional gain
-			   _param_ra_speed_i.get(), // Integral gain
-			   0, // Derivative gain
-			   _param_ra_speed_i.get() > FLT_EPSILON ? 1.f / _param_ra_speed_i.get() : 0.f, // Integral limit
-			   1); // Output limit
+	_pid_throttle.setGains(_param_ra_speed_p.get(), _param_ra_speed_i.get(), 0.f);
+	_pid_throttle.setIntegralLimit(_param_ra_speed_i.get() > FLT_EPSILON ? 1.f / _param_ra_speed_i.get() : 0.f);
```

Comment:
```suggestion
	_pid_throttle.setIntegralLimit(_param_ra_speed_i.get());
```
The reason this was the inverse of the integral gain was because the old pid library set the integral limit prior to the multiplication with the integral gain. The new library enforces the limit after the multiplication with the integral gain, so this is no longer necessary.
Note: I initally implemented this wrong in the rover modules, which is why it is inconsistent between the ackermann and differential/mecanum modules. There are open PRs that I made to fix this, but I will just close them since that fix is no longer necessary after this PR.

Reply:
Ah right, thanks for the correction. I remember expecting that, checking but getting confused by the previous naming of state vs. gain 🤦‍♂️ Let me change that 👍

Reply:
Fixed in place: https://github.com/PX4/PX4-Autopilot/compare/ce04bbdd61b48a2be6117ff4ce84edc2e1d975d1..9663c0e67387be0800d5e5870993bef826ba74f1#diff-8f63176046ab5b2a21e28c9508998e20d409b76e916999e75de9636d8d5973d3R51-R55